src/Microsoft.TemplateSearch.TemplateDiscovery/Nuget/NugetPackInfo.cs

```diff
@@ -7,14 +7,27 @@ namespace Microsoft.TemplateSearch.TemplateDiscovery.Nuget
 {
     public class NugetPackInfo : IDownloadedPackInfo
     {
-        public string VersionedPackageIdentity { get; set; }
+        private IPackInfo _info;
 
-        public string Id { get; set; }
+        internal NugetPackInfo(IPackInfo info, string filePath)
+        {
+            if (string.IsNullOrWhiteSpace(filePath))
```

Comment:
should the filePath be validated?  `File.Exists(filepath)` for example.

Reply:
I see no point validating here - almost 0% chance, and it will be validated when the package is read. We don't save the information where the package is located, it's intermediate info.